modules/rte_mis_state/rte_mis_state.module

```diff
@@ -200,17 +199,43 @@ function rte_mis_state_form_alter(&$form, FormStateInterface $form_state, $form_
                     $form_state->setValue('roles', ['district_admin']);
                     $form['field_location_details'] = rte_mis_state_roles_ajax_callback($form, $form_state);
                   }
+                  else {
+                    $form['account']['roles']['#options'] = $role_mappings[$current_user_role];
+                  }
                 }
                 // If current user role is `district_admin`
                 // then display roles according to the roles mapped.
                 elseif ($current_user_role = 'district_admin') {
+                  $form['#attached']['library'][] = 'rte_mis_core/disable_cshs_select';
+                  $form['field_location_details']['#attributes']['class'][] = 'location-details';
+                  $form['#attached']['drupalSettings']['role'] = 'district';
                   $form['account']['roles']['#options'] = $role_mappings[$current_user_role];
+                  $form['account']['roles']['#default_value'] = 'block_admin';
+                  $form['account']['roles']['#disabled'] = 'disabled';
```

Comment:
Why is the role filter still disabled? Same for district admin?

Reply:
This is for the user_register_form. Here we are keeping the roles field disabled.